modules/GKB/CommonUtils.pm

```diff
@@ -321,22 +319,26 @@ sub XOR {
     return ($expression1 || $expression2) && (!($expression1 && $expression2));
 }
 
-sub get_components {
-    my $composite_entity = shift;
-    
-    my @components;
-    foreach my $component (@{$composite_entity->hasComponent},
-                           @{$composite_entity->hasMember},
-                           @{$composite_entity->hasCandidate},
-                           @{$composite_entity->repeatedUnit}) {
-        push @components, $component;
-        my @sub_components = grep {defined} get_components($component);
-        if (@sub_components) {
-            push @components, @sub_components;
-        }
-    }
-    
-    return @components;
+sub get_components
+{
+	my $composite_entity = shift;
+
+	my @components;
+	if ($composite_entity)
+	{
+		foreach my $component (@{$composite_entity->hasComponent}, @{$composite_entity->hasMember},
+								@{$composite_entity->hasCandidate}, @{$composite_entity->repeatedUnit})
+		{
```

Comment:
How does this code actually evaluate? Like a double for-loop? (i.e. foreach $component (foreach $arrayElement), (foreach $arrayElement), (foreach $arrayElement), etc)).

Reply:
I don't think it's nested loops. I think it has the effect of concatenating all of the arrays into one and then looping through the resulting array. I've never seen this in any other language, all I did was reformat it.

Reply:
Definitely strikes me as a Perlism